custom_components/waste_collection_schedule/waste_collection_schedule/source/warszawa19115_pl.py

```diff
@@ -0,0 +1,146 @@
+import datetime
+from waste_collection_schedule import Collection
+
+from datetime import datetime
+import requests
+import logging
+import typing
+import re
+
+TITLE = 'Warszawa19115.pl'
+DESCRIPTION = "Source for Warsaw city garbage collection"  # Describe your source
+URL = "https://warszawa19115.pl/harmonogramy-wywozu-odpadow"    # Insert url to service homepage
+TEST_CASES = { # Insert arguments for test cases using test_sources.py script
+    "Street Name": {"street_address": "MARSZAŁKOWSKA 84/92, 00-514 Śródmieście"},
+    'Geolocation ID': {'geolocation_id': '76802934'},
+}
+
+_LOGGER = logging.getLogger(__name__)
+
+ICON_MAP = {
+    'green waste': 'mdi:leaf',
+    'recycling': 'mdi:recycle'
+}
+
+
+class SourceConfigurationError(ValueError):
+    pass
+
+
+class SourceParseError(ValueError):
+    pass
+
+
+
+class Source:
+    OC_GEOLOCATION_SEARCH_URL = 'https://warszawa19115.pl/harmonogramy-wywozu-odpadow?p_p_id=portalCKMjunkschedules_WAR_portalCKMjunkschedulesportlet_INSTANCE_o5AIb2mimbRJ&p_p_lifecycle=2&p_p_state=normal&p_p_mode=view&p_p_resource_id=autocompleteResourceURL&p_p_cacheability=cacheLevelPage&p_p_col_id=column-1&p_p_col_count=1'
+
+    OC_SESSION_URL = 'https://warszawa19115.pl/harmonogramy-wywozu-odpadow'
+    OC_CALENDAR_URL = 'https://warszawa19115.pl/harmonogramy-wywozu-odpadow?p_p_id=portalCKMjunkschedules_WAR_portalCKMjunkschedulesportlet_INSTANCE_o5AIb2mimbRJ&p_p_lifecycle=2&p_p_state=normal&p_p_mode=view&p_p_resource_id=ajaxResourceURL&p_p_cacheability=cacheLevelPage&p_p_col_id=column-1&p_p_col_count=1'
+
+    OC_RE_DATE_STR = re.compile(r'[^\s]+\s(\d{1,2}/\d{1,2}/\d{4})')
+
+    def __init__(self, street_address: typing.Optional[str] = None, geolocation_id: typing.Optional[str] = None):
+        if street_address is None and geolocation_id is None:
+            raise SourceConfigurationError('Either street_address or geolocation_id must have a value')
+
+        self._street_address = street_address
+        self._geolocation_id = geolocation_id
+
+    @property
+    def geolocation_id(self) -> str:
+        if self._geolocation_id is None:
+            payload='_portalCKMjunkschedules_WAR_portalCKMjunkschedulesportlet_INSTANCE_o5AIb2mimbRJ_name=' + self._street_address
+            geolocation_session = requests.Session()
+            geolocation_request = geolocation_session.get(self.OC_SESSION_URL)
+            geolocation_request.raise_for_status()
+            headers = {
+            'sec-ch-ua': '" Not A;Brand";v="99", "Chromium";v="101", "Google Chrome";v="101"',
```

Comment:
Please remove optional header fields and only keep the really mandatory fields (typically User-Agent). Move common header fields for the 2 requests to a single global variable.